galaxy/importer/tests/test_linters.py

```diff
@@ -104,3 +106,87 @@ def test_yamllint_fail():
         ]
         expected = [s.format(fp.name) for s in expected]
         assert sorted(result) == sorted(expected)
+
+
+ANSIBLELINT_TEST_FILE_OK = """---
+- name: Add mongodb repo apt_key
+  become: true
+  apt_key: keyserver=hkp
+"""
+
+
+def test_ansiblelint_ok():
+    try:
+        temp_root = tempfile.mkdtemp()
+        role_path = 'role/tasks'
+        full_path = os.path.join(temp_root, role_path)
+        os.makedirs(full_path)
+        file_name = os.path.join(full_path, 'main.yml')
+        with open(file_name, 'w') as fp:
+            fp.write(ANSIBLELINT_TEST_FILE_OK)
+            fp.flush()
+
+            linter = linters.AnsibleLinter()
+            linter.root = '%s/%s' % (temp_root, role_path.split('/')[0])
+            result = list(linter.check_files(['.']))
+            assert result == []
```

Comment:
better do assertions outside of with-block to avoid interfering with context managers

Reply:
Updated